File: XO_Game_mongo.py
import pymongo
client = pymongo.MongoClient("mongodb://localhost:27017/")
mydb = client["xoDB"]
gameRecords = mydb["gameRecords"]
gameMoves = mydb["gameMoves"]
board = [["-","-","-"],
         ["-","-","-"],
         ["-","-","-"]]
currentPlayer = "X"
winner = " "
gameGoing = True
Player1 = input("Enter Player 1 name: ")
Player2 = input("Enter Player 2 name: ") 
dict = {"player1": Player1 , "player2": Player2 }
x = gameRecords.insert_one(dict)
inserted_id = x.inserted_id
def printboard(board):
    print("  |0 | 1 | 2")
    print(" ----|---|--")
    print("0 |" + board[0][0] + " | " + board[0][1] + " | " + board[0][2])
    print("------------")
    print("1 |" + board[1][0] + " | " + board[1][1] + " | " + board[1][2])
    print("------------")
    print("2 |" + board[2][0] + " | " + board[2][1] + " | " + board[2][2])


def playerInput(board):
    global inserted_id
    row = int(input("Enter row number: "))
    colomn = int(input("Enter colomn number: "))
    if row >= 0 and row <= 2 and colomn >=0 and colomn <=2 and board[row][colomn] == "-":
        dict = {"Symbol": currentPlayer , "row": row , "colomn": colomn}
        
        board[row][colomn] = currentPlayer
        add_moves_relationship(inserted_id,dict)
        
    else:
        print("Invalid move! \n")
        playerInput(board)

def winCheck(board):
    global winner , gameGoing
    if board[0][0] == board[0][1] == board[0][2] and board[0][0]!= "-":
        winner = board[0][0]
    elif board[1][0] == board[1][2] == board[1][1] and board[1][0]!= "-":
        winner = board[1][0]
    elif board[2][1] == board[2][0] == board[2][2] and board[2][0]!= "-":
        winner = board[2][1]
    elif board[0][0] == board[1][0] == board[2][0] and board[0][0]!= "-":
        winner = board[0][0]
    elif board[0][1] == board[1][1] == board[2][1] and board[0][1]!= "-":
        winner = board[0][1]
    elif board[0][2] == board[1][2] == board[2][2] and board[2][2]!= "-":
        winner = board[0][2]
    elif board[0][0] == board[1][1] == board[2][2] and board[0][0]!= "-":
        winner = board[0][0]
    elif board[0][2] == board[1][1] == board[2][0] and board[2][0]!= "-":
        winner = board[2][0]
    if winner != " ":
        print("the winner is: " + winner)
        printboard(board)
        gameGoing = False
def tieCheck(board):
    global gameGoing
    if not any("-" in x  for x in board):
        print("It is a tie!")
        printboard(board)
        gameGoing = False
def add_moves_relationship(game_id,dict):
    from bson.objectid import ObjectId
    _id = ObjectId(game_id)
    dict = dict.copy()
    dict["owner_id"] = game_id
    gameMoves.insert_one(dict)

while gameGoing:
    printboard(board)
    playerInput(board)
    winCheck(board)
    tieCheck(board)
    
    if currentPlayer == "X":
        currentPlayer = "O"
    else:
        currentPlayer = "X"
    
if winner == "X":
    winner= Player1
elif winner =="O":
    winner = Player2
else:
    winner = "tie"

File: XO_Game_mongo.js
var MongoClient = require('mongodb').MongoClient;
var url = "mongodb://localhost:27017/mydb";
const ps = require("prompt-sync");
const prompt = ps();
var board = [
    ["-","-","-"],
    ["-","-","-"],
    ["-","-","-"]
];
var movesarr= [];
var currentPlayer = "X";
var winner = " ";
var gameGoing = true;
var Player1 = prompt("Enter Player1 name: ");
var Player2 = prompt("Enter Player2 name: ");
function drawBoard(board){
    console.log("  |0 | 1 | 2");
    console.log(" ----|---|--");
    console.log("0 |" + board[0][0] + " | " + board[0][1] + " | " + board[0][2]);
    console.log("------------");
    console.log("1 |" + board[1][0] + " | " + board[1][1] + " | " + board[1][2]);
    console.log("------------");
    console.log("2 |" + board[2][0] + " | " + board[2][1] + " | " + board[2][2]);
}
function playerInput(board){
    let row = prompt("Enter row: ");
    let colomn = prompt("Enter colomn: ");
    if (row >= 0 && row <= 2 && colomn >=0 && colomn <=2 && board[row][colomn] == "-"){
            let myobj = { "Symbol": currentPlayer , "row": row , "colomn": colomn};
            movesarr.push(myobj);
        board[row][colomn] = currentPlayer;
    }
    else{
        console.log("Invalid move! ");
        playerInput(board);
    }
}

function winCheck(board){
    if (board[0][0] == board[0][1]&& board[0][0] == board[0][2] && board[0][0]!= "-"){
        winner = board[0][0];
    }
    else if (board[1][0] == board[1][2]&& board[1][0] == board[1][1] && board[1][0]!= "-"){
        winner = board[1][0];
    }
    else if (board[2][1] == board[2][0]&& board[2][1] == board[2][2] && board[2][0]!= "-"){
        winner = board[2][1];
    }
    else if (board[0][0] == board[1][0]&& board[0][0] == board[2][0] && board[0][0]!= "-"){
        winner = board[0][0];
    }
    else if (board[0][1] == board[1][1]&& board[0][1] == board[2][1] && board[0][1]!= "-"){
        winner = board[0][1];
    }
    else if (board[0][2] == board[1][2]&& board[0][2] == board[2][2] && board[2][2]!= "-"){
        winner = board[0][2];
    }
    else if (board[0][0] == board[1][1]&& board[0][0] == board[2][2] && board[0][0]!= "-"){
        winner = board[0][0];
    }
    else if (board[0][2] == board[1][1]&& board[0][2] == board[2][0] && board[2][0]!= "-"){
        winner = board[2][0];
    }
    if (winner != " "){
        console.log("the winner is: " + winner)
        drawBoard(board)
        gameGoing = false
    }
}

function tieCheck(board){
    if (!board.some(rows => rows.includes("-"))){
        console.log("It is a tie!")
        drawBoard(board)
        gameGoing = false
    }
}
while (gameGoing){
    drawBoard(board);
    playerInput(board);
    winCheck(board);
    tieCheck(board);
    if (currentPlayer == "X"){
        currentPlayer = "O"
    }
    else{
        currentPlayer = "X"
    }
    
}
if (winner == "X"){
    winner= Player1
}
else if (winner =="O"){
    winner = Player2
}
else{
    winner = "tie"
}
MongoClient.connect(url, function(err, db) {
    if (err) throw err;
    var dbo = db.db("XOJS");
    var myobj = { "player1": Player1 , "player2": Player2 ,"winner":winner , "moves": movesarr };
      dbo.collection("GameRecords").insertOne(myobj, function(err, res) {
      if (err) throw err;
      console.log("1 document inserted");
      db.close();
    });
    
  });
